### ISSUE TITLE
getSelectedRegExMatches() returns wrong result

### ISSUE THREAD

**Issue by jive-tamaskuzdi** [OfficeDev/office-js#1086]
There is two result in a line for a contextual addin and it doesnt matter which one i click, the `getSelectedRegExMatches()` will return only one of them.

## Expected Behavior
I expect that `getSelectedRegExMatches()` returns exactly the same highlighted match that i clicked.

## Steps to Reproduce, or Live Example
Manifest part with regex:
```
<ExtensionPoint xsi:type="DetectedEntity">
              <Label resid="contextLabel" />
              <SourceLocation resid="detectedPhoneNumbersUrl" />
              <Rule xsi:type="RuleCollection" Mode="And">
                <Rule xsi:type="RuleCollection" Mode="Or">
                  <Rule xsi:type="ItemIs" ItemType="Message" />
                  <Rule xsi:type="ItemIs" ItemType="Appointment" />
                </Rule>
                <Rule xsi:type="ItemHasRegularExpressionMatch" RegExName="MyResult" RegExValue="(?:(?:(?:00 ?\d{2}|\+\d{2})|(?:\+?\d{2} ?(?:\(0\)|0))|0|(?:\+?1)) ?)?(?:(?:\(\d{2,3}\))|(?:\d{2,3}))(?:[- ])?(?:\d{3})(?:[- ])?(?:\d{3,4})" PropertyName="BodyAsPlaintext"/>
              </Rule>
            </ExtensionPoint>
```

Line in email where both numbers are highlighted but the first one is returned always:

```M: 516-821-1090 / Fax: 516-812-0100```

## Context
This prevents us to integrate a phone utility with outlook addin that makes numbers clickable.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office Online]: __Mac ___
* Host [Excel, Word, PowerPoint, etc.]: ___Outlook___
* Office version number: __16.35 (20030802)____
* Operating System: __OSX Catalina____
* Browser (if using Office Online): ____

**Comment by Rick-Kirkham** [comment]
Please provide some more information. Which host are you working with (Excel? Word?, etc.)? What is the parent object of the `getSelectedRegExMatches` method? What is the programming scenario; that is, what are you trying to accomplish?

**Comment by exextoc** [comment]
I tried the `Office.context.mailbox.item.getSelectedRegExMatches()` API with the provided ExtensionPoint XML and email line in _Outlook Desktop on Windows_ and _Outlook on the Web_.

In both clients, it returned the correct match (the one that was selected to launch the add-in). For example: `{"MyResult":["516-812-0100"]}`

Which version number of Outlook Desktop were you testing with?

And for Outlook on the Web, was the user on an on-premise Exchange server (if so, which version), or on Office 365?

**Comment by jive-tamaskuzdi** [comment]
Thanks for looking into it, let me provide more details @exextoc !
I can confirm that the issue is not present in the web Office365, it was caused by another issue, i will do a separate report for that.

Trying from mac Catalina Microsoft Outlook desktop application is wrong:
`Office.context.mailbox.diagnostics.hostVersion` return `16.35 (20030802)`

Test mail sent exactly:
```
The wrong line:
M: 516-821-1090 / Fax: 516-812-0100
```

I have both numbers highlighted but I get `{"MyResult":["516-812-1090"]}` doesnt matter which one i click.

**Comment by jive-tamaskuzdi** [comment]
I have created a repo with description in readme:
https://github.com/jive-tamaskuzdi/outlook-regex-match-repro

**Comment by exextoc** [comment]
Hey @jive-tamaskuzdi,

Thanks for the repo with repro steps. We are able to reproduce the issue, and it has been put in our backlog. Unfortunately, we do not have any timelines to share at this point.

**Comment by exextoc** [comment]
Internal tracking id (helps to provide an update later): Office:4104226

**Comment by exextoc** [comment]
We have fixed the bug and the fix should be available in build 16.40(20070100) or later.
